src/main/java/org/apache/mesos/specification/DefaultTaskSet.java

```diff
@@ -0,0 +1,52 @@
+package org.apache.mesos.specification;
+
+import org.apache.commons.lang3.builder.ReflectionToStringBuilder;
+import org.apache.mesos.Protos;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * This class provides a default implementation of the TaskSet interface.
+ */
+public class DefaultTaskSet implements TaskSet {
+    private final String name;
+    private final List<TaskSpecification> taskSpecifications;
+
+    public static DefaultTaskSet create(
+            int count,
+            String name,
+            Protos.CommandInfo command,
+            Collection<ResourceSpecification> resources,
+            Collection<VolumeSpecification> volumes) {
+
+        List<TaskSpecification> taskSpecifications = new ArrayList<>();
+        for (int i = 0; i < count; i++) {
+            taskSpecifications.add(new DefaultTaskSpecification(name + "-" + i, command, resources, volumes));
+        }
+
+        return new DefaultTaskSet(name, taskSpecifications);
+    }
+
+    public DefaultTaskSet(String name, List<TaskSpecification> taskSpecifications) {
```

Comment:
Mark this private and add a create() method with this signature (to keep the parallel with the create() method added above)?

Alternate option: Could mark protected instead of private if needed by eg the CustomTaskSet example.